lib/player.js

```diff
@@ -0,0 +1,175 @@
+const { sleep } = require("./helpers");
+const rl = require("./rl");
+
+class Player {
+  constructor(name = null, isHuman = null, ws = null) {
+    this.name = name;
+    this.isHuman = isHuman;
+    this.ws = ws;
+  }
+  getTeam(game) {
+    return Object.keys(game.players).find((key) => game.players[key] === this);
+  }
+  getOtherPlayerTeam(game) {
+    let team;
+    Object.keys(game.players).find((key) => {
+      if (game.players[key] !== this) {
+        team = key;
+      }
+    });
+    return team;
+  }
+  _isValidGridChoice(game, choice) {
+    if (
+      isNaN(choice) ||
+      choice < 1 ||
+      choice > 9 ||
+      game.choices.includes(choice)
+    ) {
+      this.log(
+        "Please enter a valid grid number. Make sure it hasn't already been picked!"
+      );
```

Comment:
I think I would move this log line into getGridChoice(). When I see a function name that starts with `is` I expect it to return a boolean and have no side-effects.

Reply:
(you will notice that this function's body then becomes `return !(isNan(choice) || ...)`